### PR TITLE
SLI-1589 Improve UI thread usage - Round 1

### PR DESCRIPTION
**Pull request by nquinquenel**
[SLI-1589](https://sonarsource.atlassian.net/browse/SLI-1589)

[SLI-1589]: https://sonarsource.atlassian.net/browse/SLI-1589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt [hunk 1]
```diff
@@ -37,25 +36,25 @@ import org.sonarlint.intellij.ui.inlay.InlayManager
...
 import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.FixSuggestionDto
 
-class ShowFixSuggestion(private val project: Project, private val file: VirtualFile, private val fixSuggestion: FixSuggestionDto) {
+class ShowFixSuggestion(
+    private val project: Project, private val file: VirtualFile, private val fixSuggestion: FixSuggestionDto
+) {
 
     fun show() {
-        val fileEditorManager = FileEditorManager.getInstance(project)
-        val psiFile = PsiManager.getInstance(project).findFile(file) ?: return
```
Comment (eray-felek-sonarsource) [inline src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt:44]: In the documentation, it is mentioned that psi file operations should not be done on the UI thread. Why are we moving these back to the UI thread again?

Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt:44]: `PsiManager.getInstance(project).findFile(file)` requires to have read access

File: src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java [hunk 2]
```diff
@@ -78,11 +79,9 @@ public SonarLintToolWindow(Project project) {
     project.getMessageBus().connect().subscribe(ProjectBindingListenerKt.getPROJECT_BINDING_TOPIC(), this);
   }
 
-  /**
-   * Must run in EDT
-   */
   public void openReportTab(AnalysisResult analysisResult) {
-    this.<ReportPanel>openTab(SonarLintToolWindowFactory.REPORT_TAB_TITLE, panel -> panel.updateFindings(analysisResult));
+    runOnUiThread(project, ModalityState.stateForComponent(getToolWindow().getComponent()),
+      () -> this.<ReportPanel>openTab(SonarLintToolWindowFactory.REPORT_TAB_TITLE, panel -> panel.updateFindings(analysisResult)));
```
[the remark this answers is not in the document]
Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java:84]: I'm dropping this from this PR

File: src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt (hunk 1 above)
Comment (eray-felek-sonarsource) [reply src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt:44]: Yes but we don't need UI Thread access

Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt:44]: Right, can move to read access only

**Pull request closed without merging**